tests/neptunelib/test_utils.py

```diff
@@ -69,6 +74,89 @@ def test_dict(self):
         self.assertEqual([{'a': 1}], as_list({'a': 1}))
 
 
+class TestAlignChannelsOnX(unittest.TestCase):
```

Comment:
From my perspective, `align_channels_on_x` is the method we should be testing. `get_channel_name_stems`, `merge_dataframes` and `sort_df_by_columns` are implementation details.

Personally, I wouldn't test/expose them.

In any case, maybe a bit-more-complex test would be in order? In particular, somewhere where `x`s are not all `0-N`, but something like `[1, 4, 7], [2, 3, 9], [3, 4, 5]`?

Reply:
Good point.

Reply:
Can numeric channels in neptune have duplicated `x_` value under some unlikely circumstances? I would assume that it's impossible but figured I'd ask.

Reply:
Oh, actually, maybe some floats as well :)

Reply:
No, duplication is impossible. The x axis is strictly increasing.

Reply:
So [here are the lines](https://github.com/neptune-ml/neptune-lib/blob/master/neptunelib/experiment.py#L124-L130).